Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp

```diff
@@ -130,52 +135,48 @@ void CalculatePlaczekSelfScattering::exec() {
   */
 
   const auto &specInfo = inWS->spectrumInfo();
-  API::MatrixWorkspace_sptr outputWS = DataObjects::create<API::HistoWorkspace>(*inWS);
+  // prep the output workspace
+  // - use instrument information from InputWorkspace
+  // - use the bin Edges from the incident flux
+  API::MatrixWorkspace_sptr outputWS =
+      DataObjects::create<API::HistoWorkspace>(*inWS, incidentWS->getSpectrum(0).binEdges());
+  // Set outputWS unit to Wavelength
+  outputWS->getAxis(0)->unit() = incidentWS->getAxis(0)->unit();
   // The algorithm computes the signal values at bin centres so they should
   // be treated as a distribution
   outputWS->setDistribution(true);
   outputWS->setYUnit("");
   outputWS->setYUnitLabel("Counts");
```

Comment:
I think I should also remove the setYUnit(""); line above?

Reply:
I think you probably need to keep this line. I think the workspace has YUnit="Counts" and YUnitLabel="" at the point it's created. So I think the YUnit does need blanking out so that both YUnit and YUnitLabel are blank strings.

Reply:
Does ConvertUnits figure out the input Unit as Wavelength in some other way?